charts/monochart/templates/deployment.yaml

```diff
@@ -182,25 +179,26 @@ spec:
       {{- if .Values.deployment.volumes }}
         {{- toYaml .Values.deployment.volumes | nindent 6 }}
       {{- end }}
-      {{- if .Values.deployment.ephemeral.enabled }}
-      - name: scratch
+
+      {{- if .Values.deployment.persistence.enabled }}
```

Comment:
@sgcg10 since `.Values.deployment.persistence` is a new attribute you need to conditionally check if `persistence` is present in .Values.

If you try running the monochart with `values.yaml` and `deployment.enabled: true` + `persistence.enabled: true` you will face an exception:

```
Error: INSTALLATION FAILED: template: monochart/templates/deployment.yaml:183:20: executing "monochart/templates/deployment.yaml" at <.Values.deployment.persistence.enabled>: nil pointer evaluating interface {}.enabled
```

You can check the original behavior of the chart in version monochart-1.0.7 (git checkout monochart-1.0.7) to test the chart.

Reply:
This line should be  `{{- if .Values.persistence.enabled }}` you should test the presence of `.Values.deployment.persistence` with hasKey function

Reply:
Here is the dry run output with monochart 1.0.7

```
❯ helm install test -f values.yaml .  --dry-run

NAME: test
LAST DEPLOYED: Mon Sep 25 09:53:12 2023
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
HOOKS:
MANIFEST:
---
# Source: monochart/templates/pvc.yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: test-monochart
  labels:
    app: monochart
    chart: monochart-1.0.7
    heritage: "Helm"
    release: "test"
spec:
  accessModes:
    - "ReadWriteOnce"
  resources:
    requests:
      storage: "8Gi"
---
# Source: monochart/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test-monochart
  labels:
    app: monochart
    chart: monochart-1.0.7
    heritage: "Helm"
    release: "test"
spec:
  replicas: 1
  selector:
    matchLabels:
      app: monochart
      release: test
  minReadySeconds: 0
  revisionHistoryLimit: 10
  template:
    metadata:
      name: test-monochart
      annotations:
        checksum/config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
        checksum/secret: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
      labels:
        app: monochart
        release: "test"
        serve: "true"
    spec:
      terminationGracePeriodSeconds: 30
      dnsPolicy: ClusterFirst
      containers:
      - name: test
        image: scratch:latest
        imagePullPolicy: IfNotPresent
        
        
        
        ports:
        volumeMounts:
        - mountPath: "/data"
          name: storage
        
      imagePullSecrets:
      volumes:
      - name: storage
        persistentVolumeClaim:
          claimName: test-monochart
```

Reply:
@Lowess I added an additional condition and it works like a charm: 

- Without persistence block
- persistence=false
- persistence=true and ephemeral=false
- persistence=true and ephemeral=true

Reply:
> This line should be `{{- if .Values.persistence.enabled }}` you should test the presence of `.Values.deployment.persistence` with hasKey function

About trying to get **Values.persistence.enabled** I think it's not reaching that value, even in StatefulSet it's being used **.Values.statefulset.persistence.enabled ** (line 131)